FinniversKit/Sources/Recycling/GridViews/NeighborhoodProfile/NeighborhoodProfileView.swift

```diff
@@ -102,79 +125,45 @@ public final class NeighborhoodProfileView: UIView {
         resetPageControl()
         resetCollectionViewLayout()
         collectionView.reloadData()
-    }
-
-    // MARK: - Overrides
-
-    // This override exists because of how we calculate view sizes in our objectPage.
-    // The objectPage needs to know the size of this view before it's added to the view hierarchy, aka. before
-    // the collectionView itself knows it's own contentSize, so we need to calculate the total height of the view manually.
-    //
-    // All we're given to answer this question is the width attribute in `targetSize`.
-    //
-    // This implementation may not work for any place other than the objectPage, because:
-    //   - it assumes `targetSize` contains an accurate targetWidth for this view.
-    //   - it ignores any potential targetHeight.
-    //   - it ignores both horizontal and vertical fitting priority.
-    public override func systemLayoutSizeFitting(
-        _ targetSize: CGSize,
-        withHorizontalFittingPriority horizontalFittingPriority: UILayoutPriority,
-        verticalFittingPriority: UILayoutPriority
-    ) -> CGSize {
-        var height = NeighborhoodProfileView.headerSpacingTop
-        height += NeighborhoodProfileHeaderView.height(forTitle: viewModel.title, width: targetSize.width)
-        height += .spacingS + collectionViewHeight(forItemHeight: calculateItemSize().height)
 
-        if isPagingEnabled {
-            height += pageControl.intrinsicContentSize.height + .spacingS
-        } else {
-            height += .spacingM
+        if let banner = viewModel.banner {
+            setupBanner(with: banner)
```

Comment:
In case we need to reconfigure this view and don't get a banner model we should hide `bannerView`. By doing these changes we can also remove the method `setupBanner(with:)`.

```swift
if let banner = viewModel.banner {
    bannerView.text = banner.text
    bannerView.buttonText = banner.link.title
    bannerView.isHidden = false
    containerStackView.setCustomSpacing(.spacingM, after: pageControl)
} else {
    bannerView.isHidden = true
    containerStackView.setCustomSpacing(.spacingS, after: pageControl)
}
```

```swift
// These three lines already exists, but add the banner view below this.
if isPagingEnabled {
    containerStackView.addArrangedSubview(pageControl)
}

containerStackView.addArrangedSubview(centeredContainerView(with: [bannerView]))
```